### PR TITLE
Set nonces manually for Ethereum transactions

### PR DESCRIPTION
**Pull request by max-sidorov**

### CODE REVIEW

File: commander/tracker/sender.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+func (t *TxsSender) sendRequest(request *txRequest) (*types.Transaction, error) {
+	tx, err := t.sendTransaction(request.params)
+	request.responseChannel <- txRequestResponse{Tx: tx, Err: err}
+	return tx, err
+}
+
+func (t *TxsSender) sendTransaction(params interface{}) (*types.Transaction, error) {
+	switch typedParams := params.(type) {
+	case *submitTransfersBatchRequestParams:
```
Comment (Marik-D) [inline commander/tracker/sender.go:68]: I'm wondering if it wouldn't be easier to encode transactions much earlier - before they are pushed through the channel. This way you'd only have a single generic transaction type with data, value, and recipients address and you wouldn't need to keep repetitive boilerplate for every transaction that we send.

Try seeing if there's a good way to utilise smart-contract bindings to just encode transactions without sending them, and if not, you can try using the raw ABIs for that.

**Comment by Marik-D** [comment]
Also let's consider merging tx sender into the client. I assume that we wouldn't want to send transactions bypassing the sender anywhere in the commander code?

**Comment by b-tarczynski** [comment]
> Also let's consider merging tx sender into the client. I assume that we wouldn't want to send transactions bypassing the sender anywhere in the commander code?

Agree with @Marik-D on that. We should merge `TxsSender` into `Client` to simplify txs sending flow.

File: commander/tracker/txs_tracker.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			continue
+		case tx := <-t.TxsChan:
+			sentTxs = append(sentTxs, tx)
+			continue
+		default:
+			break
+		}
+
+		if len(sentTxs) == 0 {
+			time.Sleep(time.Millisecond * 100)
```
Comment (b-tarczynski) [inline commander/tracker/txs_tracker.go:45]: We block receiving sent txs for 100ms what can cause blocking `sendRequest` function. That ultimately can lead to blocking rollupLoop. We should append sent tx to `sentTxs` as fast as we can.

File: commander/tracker/txs_tracker.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		}
+		if !isMined {
+			time.Sleep(time.Millisecond * 100)
+			continue
+		}
+		err = t.checkTxForFail(sentTxs[0], receipt)
+		if err != nil {
+			panic(err)
+		}
+		sentTxs = sentTxs[1:]
```
Comment (b-tarczynski) [inline commander/tracker/txs_tracker.go:61]: It's look like `WaitToBeMined` with additional `checkTxForFail` function and . I would reuse that function, especially that this logic shouldn't block receiving sent txs. I would call function with `WaitToBeMined` in separate goroutine with first tx and use some kind of error channel to return errors from this function and once it's done we should call that function with next tx.

Comment (max-sidorov) [reply commander/tracker/txs_tracker.go]: I split the tracker into two workers, and now creating additional goroutines makes no sense.

**Pull request merged**